File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    productExceptSelf(nums) {
        let allProduct = 1
        let zeroCount = 0   // count zeros, not just has0
        const result = []
        for (const num of nums) {
            if (num === 0) {
                zeroCount++
                continue
            }
            allProduct *= num
        }
        for (const num of nums) {
            if (num === 0) {
                result.push(zeroCount >= 2 ? 0 : allProduct)
                continue
            }
            result.push(zeroCount >= 1 ? 0 : allProduct / num)
        }
        return result
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.go
func containsNearbyDuplicate(nums []int, k int) bool {
	dict := make(map[int]int)
	for R, num := range nums {
		if L, ok := dict[num]; ok && R-L <= k {
			return true
		}
		dict[num] = R
	}
	return false
}
